app/controllers/admin/customers_controller.rb

```diff
@@ -129,6 +130,16 @@ def tag_rule_mapping
     def customer_tags_by_id
       BatchTaggableTagsQuery.call(Customer.of(managed_enterprise_id))
     end
+
+    def customer_already_exists?
+      Customer.exists?(email: @customer.email)
+    end
+
+    def set_created_manually_flag
+      return if @customer.email.blank?
+
+      Customer.where(email: @customer.email).update_all(created_manually: true)
+    end
```

Comment:
A user can be a customer with several enterprises. For each enterprise there's a separate customer record. So we need to check by email and enterprise_id.